### ISSUE TITLE
Time delay loading snowlines

### ISSUE THREAD

**Issue by lekah** [JamesRunnalls/frontend-snowlines#4]
First user experience shows a time delay of 1-2s to show the snowline when loading the page for the first time, while the underlying map shows immediately.
 - If it is problem of file size, should we compress the snowlines to reduce traffic?
 - Is the delay because it has to ask the S3 first for the DB and then for the snowline? If so, could the first loading be sped up by having the latest snowline lying ready on file storage somewhere?  (i.e. a snowline_default.json that we overwrite with every upload)

**Comment by JamesRunnalls** [comment]
I just checked and the issue is the size of the files. We can look at ways to compress the file or we could also break it into pieces such that it loads progressively rather than in one go

**Comment by lekah** [comment]
Ok, if we go for compression, I just checked for the latest snowline how well it compresses:
 - original: 4.4Mb
 - gzip:  544K
 - bzip2: 368K
 - xz:  373K
In short, this looks quite promising. Are any of these compression working for the frontend?

**Comment by lekah** [comment]
Also, the precision of the numbers we store is quite large, ~16 digits. We can use our custom JSONEncoder when producing the snowlines, and encode with a smaller number of significant digits, which should only have a negligible difference on the drawn snowline but reduce the size significantly.

**Comment by lekah** [comment]
The changes introduced in https://github.com/lekah/snowline/pull/3 can reduce the size of files to 1.4Mb with no significant loss of accuracy. However, compression would improve things further.
With gzip, the file of a single snowline can become 275Kb.

**Comment by JamesRunnalls** [comment]
The file size reduction has already had a significant impact on performance :)

**Comment by lekah** [comment]
Yes, I was surprised how far we could get with those simple solutions. IMO, this issue becomes less urgent to solve in the near future. However, I'd still keep it open and start thinking of and testing long-term solutions to file sizes.
In the future, the plan is to increase the resolution to maybe even 20m/pixel, which could mean that the lines might become 15x denser, since current resolution is 300m.  So this issue of file size will come back.
And maybe compression is easy to implement? 275kb is still significantly smaller than 1.4Mb
